sycl/include/sycl/handler.hpp

```diff
@@ -1047,7 +1047,7 @@ class __SYCL_EXPORT handler {
   ///
   /// Stores information about kernel properties into the handler.
   template <
-      typename KernelName,
+      bool IsESIMDKernel,
```

Comment:
The optimization needs to be described here. Otherwise, it's very tempting to restore the code on the left.

Reply:
Added a comment in 2cf3a3bd5e1c923ea0f011dc3a88b8be5a1056e7